pkg/operator/apiserver/controller/workload/workload.go

```diff
@@ -209,15 +210,23 @@ func (c *Controller) updateOperatorStatus(ctx context.Context, previousStatus *o
 
 		// we are degraded, not available and we are not progressing
 
-		deploymentDegradedCondition.Status = operatorv1.ConditionTrue
-		deploymentDegradedCondition.Reason = "PreconditionNotFulfilled"
-		deploymentDegradedCondition.Message = message
+		deploymentDegradedCondition = deploymentDegradedCondition.
+			WithStatus(operatorv1.ConditionTrue).
+			WithReason("PreconditionNotFulfilled").
+			WithMessage(message)
 
-		deploymentAvailableCondition.Status = operatorv1.ConditionFalse
-		deploymentAvailableCondition.Reason = "PreconditionNotFulfilled"
+		deploymentAvailableCondition = deploymentAvailableCondition.
+			WithStatus(operatorv1.ConditionFalse).
+			WithReason("PreconditionNotFulfilled")
 
-		deploymentProgressingCondition.Status = operatorv1.ConditionFalse
-		deploymentProgressingCondition.Reason = "PreconditionNotFulfilled"
+		deploymentProgressingCondition = deploymentProgressingCondition.
+			WithStatus(operatorv1.ConditionFalse).
+			WithReason("PreconditionNotFulfilled")
+
+		workloadDegradedCondition = workloadDegradedCondition.
+			WithStatus(operatorv1.ConditionTrue).
+			WithReason("PreconditionNotFulfilled").
```

Comment:
we don't know if the workload is degraded but because now we always have to set all conditions we need to do something about it.

maybe we should set the status to `Unknown`?

Reply:
don't have a strong opinion either. It's not a big deal because it can be easily changed. Changing the type is much harder, but we can adjust it in the future.